File: src/main/java/com/example/demo/repository/BikeRepository.java
package com.example.demo.repository;

import java.util.Optional;

import org.springframework.data.jpa.repository.JpaRepository;

import com.example.demo.model.Bicycle;

public interface BikeRepository extends JpaRepository<Bicycle, Long> {
    Optional<Bicycle> findBySerialNumber(String serialNumber);
}

File: src/main/java/com/example/demo/repository/BikeRepository.kt
package com.example.demo.repository

import com.example.demo.model.Bicycle
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface BikeRepository : JpaRepository<Bicycle, Long> {
    fun findBySerialNumber(serialNumber: String?): Bicycle?
}
